universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -122,8 +125,53 @@ public boolean equals(@Nullable Object object) {
     return false;
   }
 
+  public boolean hasPath(Path path) {
+    try {
+      return read(path, Object.class).isPresent();
+    } catch (JsonPathTypeMismatchException e) {
+      return false;
+    }
+  }
+
   @Override
   public int hashCode() {
     return Objects.hash(jsonData.jsonString());
   }
+
+  public List<Path> mergeFrom(ApplicantData other) {
+    Map<?, ?> rootAsMap = other.jsonData.read("$", Map.class);
+    return mergeFrom(Path.empty(), rootAsMap);
+  }
+
+  private List<Path> mergeFrom(Path rootKey, Map<?, ?> other) {
+    List<Path> pathsRemoved = new ArrayList<>();
```

Comment:
(blocking) this method should return an `ImmutableList`, and implemented with `ImmutableList.Builder`.

Reply:
also if its just for testing, it could be `ImmutableSet`.